core/src/main/scala/org/scalatra/control.scala

```diff
@@ -25,6 +29,9 @@ trait Control {
     headers: Map[String, String] = Map.empty,
     reason: String = null
   ): Nothing = {
+    if ((reason != null && reason != "") && body.!=((): Unit))
+      logger.warn("Calls that set both reason and body as arguments to halt method will be dropped in Scalatra 2.7.0!!")
```

Comment:
Is it impossible to handle as follows?

- If body is ommitted then use `sendError()`
- If body is specified then use `setStatus()` and write body in Scalatra

This means that reason will be deprecated, but body is not.